xarray/core/datatree_mapping.py

```diff
@@ -13,25 +14,38 @@
 
 
 @overload
-def map_over_datasets(func: Callable[..., Dataset | None], *args: Any) -> DataTree: ...
+def map_over_datasets(
+    func: Callable[
+        ...,
+        Dataset | None,
+    ],
+    *args: Any,
+    kwargs: Mapping | None = None,
```

Comment:
Should technically be `Mapping[str, Any]` shouldn't it? Because you can't pass a kwarg value via a non-str key.